fedimint-testing/src/btc/mock.rs

```diff
@@ -274,6 +275,17 @@ impl BitcoinTest for FakeBitcoinTest {
             return fee;
         }
     }
+
+    async fn get_tx_block_height(&self, txid: &Txid) -> Option<u64> {
+        self.inner
+            .read()
+            .expect("RwLock poisoned")
+            .blocks
+            .clone()
```

Comment:
Why clone everything? That's going to take longer than the search itself.

Reply:
Really, could add a new `tx_to_block_height: BTreeMap<TxId, BlockHeight>;`, maintaine it when adding blocks and be able to make a O(1) lookup here.

Reply:
Good callout. I went with the `txid_to_block_height` approach and refactored the other method we're unnecessarily cloning all blocks (original inspiration that I didn't scrutinize).